subcommand/acl-init/command.go

```diff
@@ -45,6 +46,8 @@ func (c *Command) init() {
 		"Name of Kubernetes namespace where the servers are deployed")
 	c.flags.StringVar(&c.flagACLDir, "acl-dir", "/consul/aclconfig",
 		"Directory name of shared volume where acl config will be output")
+	c.flags.StringVar(&c.flagTokenSinkFile, "token-sink-file", "",
+		"Filepath to write acl token to when -init-type is sync")
```

Comment:
I was always confused about why we're calling the non-client type `sync`. It doesn't look like it's used anywhere in the command (except for the new condition you're adding). Do you think it's worth removing it?

Reply:
From commit history I think it's because this was previously only used by catalog sync:
```
2) acl-init This is run by the init container of clients and (optionally) the catalog sync pod. It waits for a Kubernetes secret to be populated with an ACL token that will be consumed and applied to the pod.
```

Options:
1. Ignore `-init-type` unless it's `client`. Change Helm to not pass `-init-type=sync`. This will be backwards compatible.
1. Add new `-init-type=wait`?

Reply:
I like option 1. it seems like init-type client just adds to whatever command does by default (i.e. write client config), so to me that option seems the most intuitive.